operator/api/v1alpha1/watcher_types.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const ManagedBylabel = "operator.kyma-project.io/managed-by"
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// WatcherSpec defines the desired state of Watcher.
+type WatcherSpec struct {
+	// ServiceInfo describes the service information of the operator
+	ServiceInfo Service `json:"serviceInfo"`
+
+	// LabelsToWatch describes the labels that should be watched
+	LabelsToWatch map[string]string `json:"labelsToWatch"`
+
+	// Field describes the subresource that should be watched
+	// Value can be one of ("spec", "status")
+	Field FieldName `json:"field"`
```

Comment:
please add `+kubebuilder:validation:Enum`

Reply:
had not pushed the crd until now, it's also in